### PR TITLE
feat: new offer uuid and fingerprint

### PR DESCRIPTION
**Pull request by h4sh3d**
Switch behavior from offer id and public offer id computed as hash of serialized content to standalone `uuid` field in `Offer` type.

A new function `.fingerprint()` is introduced to match the previous behavior of `.id()`.

### CODE REVIEW

File: tests/negotiation.rs [hunk 1]
```diff
@@ -129,47 +123,70 @@ fn serialize_public_offer() {
...
-                   0040a0000000108140000000000000002";
+    let inval = "47435357415001004450e567b1106f429247bb680e5fe0c80200000080800000800800a08601000\
+                 00000000800c80000000000000004000a00000004000a0000000108001400000000000000022100\
+                 03b31a0a70343bb46f3db3768296ac5027f9873921b37f852860c690063ff9e4c90000000000000\
+                 0000000000000000000000000000000000000000000000000000000260700";
     let pub_offer: Result<
         PublicOffer<bitcoin::Amount, monero::Amount, CSVTimelock, SatPerVByte>,
         consensus::Error,
-    > = deserialize(&hex::decode(invalid).unwrap()[..]);
+    > = deserialize(&hex::decode(inval).unwrap()[..]);
```
Comment (TheCharlatan) [inline tests/negotiation.rs:179]: invalid?

**Review by TheCharlatan — CHANGES_REQUESTED** [review]
Very nice change :)

**Comment by TheCharlatan** [comment]
Is there something we can do to get strict encoding for the UUID alone? Maybe wrap it in a struct?

**Comment by h4sh3d** [comment]
> Is there something we can do to get strict encoding for the UUID alone? Maybe wrap it in a struct?

I thought about it. It is not the nicest but `Uuid` is fully compatible with `[u8; 16]`, so I'd try using that instead when strict-encoding is needed, or wrap it in node. I don't have a better solution yet.

File: tests/negotiation.rs (hunk 1 above)
Comment (h4sh3d) [reply tests/negotiation.rs]: Was to keep the same number of chars as 'valid' for the alignment :)

Done in dae445940260b6008766f1edc2e4de01bb769fdc

**Review by TheCharlatan — APPROVED** [review]

**Pull request merged**